java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfo.java

```diff
@@ -18,73 +18,128 @@
 
 import sleeper.core.statestore.FileInfo;
 
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
-
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A data structure used by the {@link S3FileInfoStore} to represent files and their references.
+ * Internal references are created by splitting compactions and deleted by standard compactions.
```

Comment:
I think this would be a bit clearer:
'Internal references refer to file references that are added by an ingest process or created by compactions'

Reply:
Thanks, updated the javadoc